test/state-registry/stateRegistry.js

```diff
@@ -0,0 +1,849 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const fs = require('fs');
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+
+const getRandomInt = function (min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+};
+
+const serializeSmartContractResponse = (arr) => {
+  return JSON.stringify(arr, (_, value) => {
+    if (typeof value === 'bigint') {
+      return Number(value);
+    }
+    return value;
+  });
+};
+
+const STATE_OBJECT_DIR = './test/state-registry/states.json';
+const CONTRACT_NAME = 'StateRegistry';
+
+describe('@migration States Tests', () => {
+  describe('@pre-migration States Tests', () => {
+    const ITERATIONS = 1;
+    let contract;
+    let statesObject = {};
+
+    before(async function () {
+      signers = await ethers.getSigners();
+
+      const contractFactory = await ethers.getContractFactory(CONTRACT_NAME);
+      contract = await contractFactory.deploy();
+      statesObject['contract_address'] = contract.target;
+    });
+
+    after(() => {
+      fs.writeFileSync(STATE_OBJECT_DIR, JSON.stringify(statesObject));
+    });
+
+    for (let i = 0; i < ITERATIONS; i++) {
+      describe('should test value types', async function () {
+        it('should test bool', async function () {
+          const randomBool = getRandomInt(0, 1) > 0;
+
+          const tx = await contract.setVarBool(randomBool);
+          await tx.wait();
+
+          const resp = await contract.getVarBool();
+          statesObject['VarBool'] = resp;
```

Comment:
For sets like this are we duplicating the action every time the i loop runs?
If so do tests like "should test bool" and  "should test address" be inside the loop?